lib_xua/src/dfu/dfu.xc

```diff
@@ -260,11 +250,46 @@ static int DFU_Upload(unsigned int request_len, unsigned int block_num, unsigned
     return 64;
 }
 
+
+static unsigned transition_dfu_download_state()
+{
+    int tries = 0;
+    if(!flash_cmd_start_write_image_in_progress) // If flash_cmd_start_write_image() is done, transition to IDLE since the actual flash writes (flash_cmd_write_page_data) are synchronous
+    {
+        return STATE_DFU_DOWNLOAD_IDLE;
+    }
+    else
+    {
+        while(flash_cmd_start_write_image_in_progress && (tries < 50)) // Erase multiple sectors in one GET_STATUS call
```

Comment:
would be nice to tidy up this 50 magic number. const int total_tries = 50 or something.

Reply:
done